assets/src/apps/authoring/components/RightMenu/RightMenu.tsx

```diff
@@ -234,8 +239,13 @@ const RightMenu: React.FC<any> = () => {
         // schema
         if (instance.getSchema) {
           const customPartSchema = instance.getSchema();
+          let requiredFields = [];
+          if(instance.getRequiredFields){
```

Comment:
do we need to have a separate method for this? I'm pretty sure that `required` *is* part of the json schema format already, so we can just include it in the schema directly for any object. alternatively we could just consider any property that is in the provided schema to be "required" for the purposes of this json editor

Reply:
also the schema in the individual components is *just* for the custom section, externally we are requiring that `id` and `type` is present, and we can set that in the part schema. also, since we're sort of moving size and position information into that "common" schema (even though they still end up living under the `custom` property for now, we can handle them being required outside as well. speaking of, we should probably move palette into the common section and have it not be something that each component needs to declare, but we dont' have to do that during this pr

Reply:
In the schema, we are getting properties section of the actual schema, whereas the required array is at the level of properties in schema json.... so i had to put separate method for that.

Reply:
each one doesn't even do ID though, I'm saying just put required in the partSchema (part.ts) for the id, and then for the component itself (at least for the purposes of the json editor) just dynamically grab all the keys and call them required so that they are forced to match the component (do that in the editor not the component)

Reply:
ok.. so part of the suggested changes are made.... but the whole required fields functionality is not addded in this PR.
I have created a different ticket for that... so I will create a diff PR for this requirement.